Feature/11863 fss tooling implement set expiry date

rvijayavel opened the pull request:
This PR contains -
 Set expiry date implementation & related UTs
Added validations for expiry date in new batch job and UT
Moved macro into common place
Dotnet client backward compatibility related changes ( #12460)

jimjamjahaa commented on FileShareService.DesktopClient.Core/Jobs/SetExpiryDateJob.cs:
```diff
@@ -1,27 +1,53 @@
+            JToken? expiryDateToken = jsonToken.SelectToken("actionParams.expiryDate");
+
+            //Set value if key exists
+            IsExpiryDateKeyExist = expiryDateToken != null;
```
Rename variable from `IsExpiryDateKeyExist` to `ExpiryDateKeyExists` please

rvijayavel replied:
Done rename to ExpiryDateKeyExists as suggested.

jimjamjahaa reviewed (changes requested):
A very small thing, just a variable name change. 

Otherwise looks good.

jimjamjahaa reviewed (approved):
lgtm

bhalcha13813 reviewed (approved):
looks good

crdandoukho reviewed (changes requested):
Hi Vijay, 
Can you please check your changes have not resulted in less error information being reported to the user when a json file with invalid file structure is loaded into the app.
Please find attached screen captures showing the error message displayed for sampleActionsWithInvalidFormat.json in main and this branch. I think this branch is not displaying the error message correctly, could you please check?
I have also noticed in this branch the top error message is not cleared when reloading a different json file. Could you please address this too?
<img width="339" alt="sampleActionsWithInvalidFormat_main" src="https://user-images.githubusercontent.com/91598629/146033071-4aaef550-ece4-44f7-b395-61f65e4533bd.PNG">
<img width="346" alt="sampleActionsWithInvalidFormat_11860" src="https://user-images.githubusercontent.com/91598629/146033106-26daa354-866a-400b-b6b5-24709d64a431.PNG">

rvijayavel commented:
> Hi Vijay, Can you please check your changes have not resulted in less error information being reported to the user when a json file with invalid file structure is loaded into the app. Please find attached screen captures showing the error message displayed for sampleActionsWithInvalidFormat.json in main and this branch. I think this branch is not displaying the error message correctly, could you please check? I have also noticed in this branch the top error message is not cleared when reloading a different json file. Could you please address this too? <img alt="sampleActionsWithInvalidFormat_main" width="339" src="https://user-images.githubusercontent.com/91598629/146033071-4aaef550-ece4-44f7-b395-61f65e4533bd.PNG"> <img alt="sampleActionsWithInvalidFormat_11860" width="346" src="https://user-images.githubusercontent.com/91598629/146033106-26daa354-866a-400b-b6b5-24709d64a431.PNG">

@crdandoukho - This is missed from my side. Fixed it and modified the UT to check the error count also.

rvijayavel commented:
@crdandoukho - This is missed from my side. Fixed it and modified the UT to check the error count also.
